compiler/src/codegen/compcore.re

```diff
@@ -2054,7 +2077,21 @@ let compile_prim1 = (wasm_mod, env, p1, arg): Expression.t => {
             Expression.Const.make(wasm_mod, const_false()),
           ),
           AssertionError,
-          [],
+          [
+            () => ([], MImmConst(MConstI32(Int32.of_int(loc.Grain_parsing.Location.loc_start.pos_lnum)))),
+            () => {
+              let set_fname = set_swap(
+                ~ty=I32Type,
+                wasm_mod,
+                env,
+                0,
+                allocate_string(wasm_mod, env, loc.Grain_parsing.Location.loc_start.pos_fname)
+              );
+              ([
+                Expression.Drop.make(wasm_mod, set_fname),
+               ], MImmBinding(MSwapBind(Int32.zero, I32Type)))
+            }
+          ],
```

Comment:
You shouldn't need the swap or the delayed functions here; this should be able to just be
```suggestion
          [
            MImmConst(MConstI32(Int32.of_int(loc.Grain_parsing.Location.loc_start.pos_lnum))),
            allocate_string(wasm_mod, env, loc.Grain_parsing.Location.loc_start.pos_fname)
          ],
```
(and updating `call_error_handler` to just take regular args)

Reply:
The problem is that `MImmConst` is an immediate, when `allocate_string` is a Binaryen expression. I did try to clean it up (just pushed), but it is still giving me the same issue as I described on Discord.